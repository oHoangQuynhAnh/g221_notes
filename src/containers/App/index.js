import React from 'react'
import { BrowserRouter, Switch } from 'react-router-dom'

import routes from '../../routes'
import DefaultLayout from '../../layouts/DefaultLayout'
import HomePage from '../HomePage'
import LoginPage from '../LoginPage'
import DummyPage from '../DummyPage'

export default function App() {
  return (
    <BrowserRouter>
      <Switch>
        <DefaultLayout exact path={routes.home} component={HomePage} />
        <DefaultLayout path={routes.login} component={LoginPage} />
        <DefaultLayout path={routes.dummy} component={DummyPage} />
      </Switch>
    </BrowserRouter>
  )
}
