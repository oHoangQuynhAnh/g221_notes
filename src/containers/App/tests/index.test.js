import React from 'react'
import { render } from '@testing-library/react'

import App from '../index'

describe('<App>', () => {
  it('should render and match the snapshot', () => {
    const { container } = render(<App />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
