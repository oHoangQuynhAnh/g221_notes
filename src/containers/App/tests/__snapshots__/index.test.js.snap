// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> should render and match the snapshot 1`] = `
.c1 {
  display: grid;
  grid-template-columns: 2fr 8fr 2fr;
}

.c2 {
  grid-column-start: 3;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin: 0 auto;
}

.c3 {
  margin-top: 60px;
}

<div
  class="c0"
>
  <header
    class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular c1 MuiToolbar-gutters"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Home
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="c2"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Signup
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
  <div
    class="MuiContainer-root c3 MuiContainer-maxWidthSm"
  >
    <p>
      Homepage
    </p>
  </div>
</div>
`;
